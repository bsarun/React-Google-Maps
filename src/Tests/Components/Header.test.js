import React from 'react';
import Header from '../../Components/Header';
import { IntlProvider } from 'react-intl';
import i18n from '../../i18n.en.json';
import renderer from 'react-test-renderer';
import {shallow, mount, render} from 'enzyme';
import LayoutActions from '../../Actions/LayoutActions';

describe('(Component) Header', () => {
	it('testing Header ', () => {
		var output = renderer.create(<IntlProvider locale="en" messages={i18n} ><Formsy.Form><Header localeMenuIsOpen={true} fabIsOpen={true} showSaveAndExit={true} showLogOutUser={true} className="test"/></Formsy.Form></IntlProvider>);
		let tree = output.toJSON();
		expect(tree).toMatchSnapshot();	
	});
	it('calls "handleSaveClick()" on button click - using prototype', () => {
		const mockedEvent = { target: {} ,preventDefault:()=>{}}; 
		const wrapper = mount(<IntlProvider locale="en" messages={i18n} ><Formsy.Form><Header localeMenuIsOpen={true} fabIsOpen={true} showSaveAndExit={true} showLogOutUser={false} className="test"/></Formsy.Form></IntlProvider>);
		wrapper.find('#save-exit-btn').simulate('click', mockedEvent );
		//expect(wrapper.find('HTMLButtonElement')).toHaveLength(1);
	});

	it('calls "handleLogOutUserClick()" on button click - using prototype', () => {
		const mockedEvent = { target: {} ,preventDefault:()=>{}}; 
		const wrapper = mount(<IntlProvider locale="en" messages={i18n} ><Formsy.Form><Header localeMenuIsOpen={true} fabIsOpen={true} showSaveAndExit={false} showLogOutUser={true} className="test"/></Formsy.Form></IntlProvider>);
		wrapper.find('#logout-btn').simulate('click', mockedEvent );
		//expect(wrapper.find('HTMLButtonElement')).toHaveLength(1);
	});

});