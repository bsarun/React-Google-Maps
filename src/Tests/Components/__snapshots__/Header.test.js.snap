// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Header testing Header  1`] = `
<form
  onSubmit={[Function]}
>
  <header
    className="main-header has-save-and-exit test"
    role="banner"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-3 col-sm-2"
        >
          <h1
            className="logo"
          >
            Covered California
          </h1>
        </div>
        <div
          className="btn-column col-xs-9 col-sm-10 text-right"
        >
          <div
            className="locale-menu open en"
          >
            <button
              className="btn btn-toggle hidden-sm hidden-md hidden-lg"
              onClick={[Function]}
            >
              <span
                className="sr-only"
              >
                <span>
                  Open language menu
                </span>
              </span>
            </button>
            <form
              className="locale-options"
              onSubmit={[Function]}
            >
              <div
                className="form-group segmented-control"
              >
                
                
                <div
                  className="controls"
                >
                  <span>
                    <button
                      className="segmented-option btn selected btn-primary"
                      data-value="en"
                      disabled={false}
                      id="locale-options-0"
                      onBlur={[Function]}
                      onClick={[Function]}
                    >
                      <span
                        className="wrapper-text"
                      >
                        English
                      </span>
                    </button>
                  </span>
                  <span>
                    <button
                      className="segmented-option btn btn-default"
                      data-value="es"
                      disabled={false}
                      id="locale-options-1"
                      onBlur={[Function]}
                      onClick={[Function]}
                    >
                      <span
                        className="wrapper-text"
                      >
                        Espaï¿½ol
                      </span>
                    </button>
                  </span>
                </div>
                
                
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </header>
</form>
`;
